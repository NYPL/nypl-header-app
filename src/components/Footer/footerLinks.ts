const link = {
  nyplLinks: [
    [
      {
        text: "Accessibility",
        href: "http://www.nypl.org/accessibility",
      },
      {
        text: "Press",
        href: "http://www.nypl.org/press",
      },
      {
        text: "Careers",
        href: "http://www.nypl.org/careers",
      },
      {
        text: "Space Rental",
        href: "http://www.nypl.org/spacerental",
      },
    ],
    [
      {
        text: "Privacy Policy",
        href: "http://www.nypl.org/help/about-nypl/legal-notices/privacy-policy",
      },
      {
        text: "Other Policies",
        href: "http://www.nypl.org/policies",
      },
      {
        text: "Terms & Conditions",
        href: "http://www.nypl.org/terms-conditions",
      },
      {
        text: "Governance",
        href: "http://www.nypl.org/help/about-nypl/leadership/board-trustees",
      },
    ],
    [
      {
        text: "Rules & Regulations",
        href: "http://www.nypl.org/help/about-nypl/legal-notices/rules-and-regulations",
      },
      {
        text: "About NYPL",
        href: "http://www.nypl.org/help/about-nypl",
      },
      {
        text: "Language",
        href: "http://www.nypl.org/language",
      },
    ],
  ],
  socialMedia: [
    {
      href: "https://www.facebook.com/nypl",
      iconName: "legacySocialFacebook",
      title: "NYPL on Facebook",
    },
    {
      href: "https://instagram.com/nypl",
      iconName: "legacySocialInstagram",
      title: "NYPL on Instagram",
    },
    {
      href: "https://www.youtube.com/user/NewYorkPublicLibrary",
      iconName: "legacySocialYoutube",
      title: "NYPL on Youtube",
    },
  ],
};

export default link;
