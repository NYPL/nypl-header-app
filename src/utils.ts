// This is used just for the app's environment value, either
// qa or production. This function is needed to get around jest
// throwing an error trying to load a cjs module (and this syntax
// is also Vite-specific).
export const getEnvVar = (key: string) => {
  return import.meta.env[key];
};
