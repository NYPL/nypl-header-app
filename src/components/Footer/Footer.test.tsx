import { render, screen, within } from "@testing-library/react";
import { axe } from "jest-axe";
import * as React from "react";
import * as renderer from "react-test-renderer";

import Footer from "./Footer";

import * as envUtils from "../../utils";

jest.mock("../../utils", () => ({
  getEnvVar: jest.fn(),
}));

describe("Footer Accessibility", () => {
  beforeAll(() => {
    (envUtils.getEnvVar as jest.Mock).mockImplementation((key) =>
      key === "VITE_APP_ENV" ? "qa" : ""
    );
  });

  it("passes axe accessibility", async () => {
    const { container } = render(<Footer />);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Footer", () => {
  beforeAll(() => {
    (envUtils.getEnvVar as jest.Mock).mockImplementation((key) =>
      key === "VITE_APP_ENV" ? "qa" : ""
    );
  });

  beforeEach(() => {
    render(<Footer />);
  });

  it("renders a footer element container", () => {
    expect(screen.getByRole("contentinfo")).toBeInTheDocument();
  });

  it("renders a list of three link lists", () => {
    const list = screen.getByTestId("text-links");
    const innerLists = within(list).getAllByRole("list");

    expect(innerLists).toHaveLength(3);
    expect(within(innerLists[0]).getAllByRole("link")).toHaveLength(4);
    expect(within(innerLists[0]).getAllByRole("link")[0]).toHaveTextContent(
      "Accessibility"
    );
    expect(within(innerLists[1]).getAllByRole("link")).toHaveLength(4);
    expect(within(innerLists[1]).getAllByRole("link")[0]).toHaveTextContent(
      "Privacy Policy"
    );
    expect(within(innerLists[2]).getAllByRole("link")).toHaveLength(3);
    expect(within(innerLists[2]).getAllByRole("link")[0]).toHaveTextContent(
      "Rules & Regulations"
    );
  });

  it("renders a list of four social media icons", () => {
    const list = screen.getByTestId("social-media-links");
    const icons = within(list).getAllByRole("img");

    expect(icons).toHaveLength(4);
    const facebookIcon = screen.getByTitle("NYPL on Facebook");
    const twitterIcon = screen.getByTitle("NYPL on Twitter");
    const instagramIcon = screen.getByTitle("NYPL on Instagram");
    const youtubeIcon = screen.getByTitle("NYPL on Youtube");
    expect(facebookIcon).toBeInTheDocument();
    expect(twitterIcon).toBeInTheDocument();
    expect(instagramIcon).toBeInTheDocument();
    expect(youtubeIcon).toBeInTheDocument();
  });

  it("renders a footer image of the NYPL building", () => {
    const nyplBuildingImage = screen.getByAltText("NYPL Main Building Facade");

    expect(nyplBuildingImage).toBeInTheDocument();
    expect(nyplBuildingImage).toHaveAttribute(
      "src",
      "https://cdn-d8.nypl.org/s3fs-public/2020-05/NYPL_MainFacadeRev2Cam2.png"
    );
  });

  it("renders the NYPL logo", () => {
    expect(
      screen.getByTitle("The New York Public Library")
    ).toBeInTheDocument();
  });

  it("renders the UI snapshot correctly", () => {
    const footer = renderer.create(<Footer />).toJSON();
    expect(footer).toMatchSnapshot();
  });
});
