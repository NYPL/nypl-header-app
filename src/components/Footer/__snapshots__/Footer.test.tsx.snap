// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the UI snapshot correctly 1`] = `
<footer
  className="css-1xdhyk6"
  id="footer"
  role="contentinfo"
>
  <div
    className="css-0"
  >
    <ul
      className="css-1xdhyk6"
      data-testid="text-links"
    >
      <li>
        <ul
          className="css-1xdhyk6"
        >
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/accessibility"
              rel={null}
              target={null}
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/press"
              rel={null}
              target={null}
            >
              Press
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/careers"
              rel={null}
              target={null}
            >
              Careers
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/spacerental"
              rel={null}
              target={null}
            >
              Space Rental
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="css-1xdhyk6"
        >
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
              rel={null}
              target={null}
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/policies"
              rel={null}
              target={null}
            >
              Other Policies
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/terms-conditions"
              rel={null}
              target={null}
            >
              Terms & Conditions
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/help/about-nypl/leadership/board-trustees"
              rel={null}
              target={null}
            >
              Governance
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="css-1xdhyk6"
        >
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/help/about-nypl/legal-notices/rules-and-regulations"
              rel={null}
              target={null}
            >
              Rules & Regulations
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/help/about-nypl"
              rel={null}
              target={null}
            >
              About NYPL
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="//www.nypl.org/language"
              rel={null}
              target={null}
            >
              Language
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="css-1xdhyk6"
      data-testid="social-media-links"
    >
      <li>
        <a
          aria-label="NYPL on Facebook"
          className="css-1xdhyk6"
          href="https://www.facebook.com/nypl"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            aria-label="NYPL on Facebook"
            className="chakra-icon css-1vgifmd"
            focusable={false}
            role="img"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              NYPL on Facebook
            </title>
            <path
              d="M11.5961 6.44469V11.0255H8.24243V16.6277H11.5961V32H18.4849V16.6277H23.107C23.107 16.6277 23.5403 13.9417 23.7495 11.0044H18.511V7.17463C18.5738 6.81581 18.7551 6.48847 19.0259 6.24504C19.2966 6.0016 19.6412 5.85615 20.0044 5.83198H23.7576V0H18.6548C11.4259 0 11.5961 5.60745 11.5961 6.44469Z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          aria-label="NYPL on Twitter"
          className="css-1xdhyk6"
          href="https://twitter.com/nypl"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            aria-label="NYPL on Twitter"
            className="chakra-icon css-1vgifmd"
            focusable={false}
            role="img"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              NYPL on Twitter
            </title>
            <path
              d="M32 6.1334C30.8013 6.66196 29.5304 7.00873 28.2293 7.16227C29.5986 6.3479 30.6248 5.06293 31.116 3.54732C29.8265 4.30883 28.4167 4.84506 26.947 5.13297C25.9614 4.07664 24.6546 3.3752 23.2296 3.1375C21.8045 2.89981 20.3409 3.13916 19.0657 3.81841C17.7906 4.49765 16.7754 5.5788 16.1776 6.89407C15.5798 8.20933 15.4328 9.68514 15.7596 11.0925C13.152 10.964 10.6005 10.29 8.26961 9.11408C5.93877 7.93815 3.88049 6.2864 2.2277 4.26552C1.38825 5.70402 1.13105 7.4092 1.50885 9.0313C1.88665 10.6534 2.87082 12.0695 4.25949 12.989C3.21742 12.9567 2.19806 12.6765 1.28584 12.1717C1.28537 12.199 1.28528 12.2263 1.28528 12.2539C1.28819 13.7657 1.81433 15.2299 2.77433 16.3978C3.73433 17.5657 5.06901 18.3653 6.55168 18.6609C5.58463 18.9223 4.57082 18.9606 3.58681 18.7728C4.00937 20.0735 4.82743 21.2099 5.92686 22.0234C7.02628 22.8369 8.35222 23.2869 9.71967 23.3106C7.39091 25.1282 4.52022 26.1129 1.56612 26.1076C1.04275 26.1077 0.519815 26.0773 0 26.0163C3.00487 27.9371 6.49752 28.9559 10.0638 28.9519C22.1396 28.9519 28.7431 18.9955 28.7431 10.3611C28.7431 10.0778 28.7368 9.79604 28.7241 9.51581C30.0085 8.59202 31.1178 7.44663 32 6.1334V6.1334Z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          aria-label="NYPL on Instagram"
          className="css-1xdhyk6"
          href="https://instagram.com/nypl"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            aria-label="NYPL on Instagram"
            className="chakra-icon css-1vgifmd"
            focusable={false}
            role="img"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              NYPL on Instagram
            </title>
            <path
              d="M32 27.8956C31.9946 28.9826 31.5604 30.0234 30.7919 30.7921C30.0234 31.5607 28.9826 31.995 27.8956 32.0007H4.10437C3.01744 31.995 1.97664 31.5607 1.20812 30.7921C0.439601 30.0234 0.00544922 28.9826 0 27.8956L0 4.10437C0.00562488 3.01755 0.439854 1.97686 1.20836 1.20836C1.97686 0.439854 3.01755 0.00562488 4.10437 0L27.8956 0C28.9824 0.00562488 30.0231 0.439854 30.7916 1.20836C31.5601 1.97686 31.9944 3.01755 32 4.10437V27.8956ZM28.3753 13.5421H25.5628C25.8392 14.4253 25.9795 15.3455 25.9789 16.2709C25.9586 17.5582 25.6849 18.8289 25.1734 20.0104C24.662 21.1918 23.9227 22.261 22.998 23.1567C22.0732 24.0524 20.9811 24.7572 19.7839 25.2308C18.5867 25.7043 17.308 25.9374 16.0207 25.9167C13.4242 25.9542 10.9189 24.9594 9.05554 23.1507C7.19216 21.342 6.1231 18.8675 6.08329 16.2709C6.08277 15.3452 6.22331 14.4248 6.50008 13.5414H3.56247V27.042C3.5614 27.2092 3.59355 27.375 3.65705 27.5297C3.72056 27.6844 3.81415 27.8249 3.93239 27.9432C4.05064 28.0614 4.19119 28.155 4.34589 28.2185C4.50059 28.282 4.66636 28.3142 4.83358 28.3131H27.1049C27.2721 28.3142 27.4378 28.282 27.5925 28.2185C27.7471 28.155 27.8876 28.0614 28.0058 27.9431C28.124 27.8249 28.2175 27.6843 28.281 27.5296C28.3444 27.3749 28.3765 27.2092 28.3753 27.042V13.5421ZM16.0207 9.70865C14.3442 9.68522 12.727 10.328 11.5239 11.4959C10.3209 12.6638 9.63053 14.2613 9.60428 15.9378C9.6307 17.6141 10.3212 19.2115 11.5242 20.3792C12.7272 21.547 14.3444 22.1897 16.0207 22.1662C16.8525 22.1802 17.6788 22.0301 18.4525 21.7246C19.2262 21.419 19.9321 20.9641 20.53 20.3857C21.1278 19.8073 21.6059 19.1168 21.9368 18.3536C22.2677 17.5904 22.445 16.7695 22.4586 15.9378C22.4451 15.106 22.2678 14.285 21.937 13.5218C21.6061 12.7585 21.1281 12.0679 20.5302 11.4894C19.9324 10.911 19.2264 10.456 18.4526 10.1504C17.6789 9.84485 16.8525 9.69474 16.0207 9.70865V9.70865ZM28.3753 5.00017C28.3744 4.61908 28.2226 4.25388 27.953 3.98454C27.6834 3.7152 27.318 3.56367 26.9369 3.56314H23.3123C22.9314 3.56402 22.5664 3.71571 22.2971 3.98501C22.0278 4.25431 21.8761 4.61931 21.8752 5.00017V8.43753C21.8758 8.81862 22.0273 9.18396 22.2966 9.45355C22.566 9.72315 22.9312 9.87501 23.3123 9.8759H26.9376C27.3188 9.87519 27.6842 9.7234 27.9537 9.45379C28.2232 9.18418 28.3748 8.81873 28.3753 8.43753V5.00017Z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          aria-label="NYPL on Youtube"
          className="css-1xdhyk6"
          href="https://www.youtube.com/user/NewYorkPublicLibrary"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            aria-label="NYPL on Youtube"
            className="chakra-icon css-1vgifmd"
            focusable={false}
            role="img"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              NYPL on Youtube
            </title>
            <path
              d="M31.3319 8.27117C31.1465 7.60661 30.7927 7.00113 30.3048 6.51325C29.8169 6.02538 29.2115 5.67161 28.5469 5.48614C26.0526 4.81789 16.0142 4.81789 16.0142 4.81789C16.0142 4.81789 6.00118 4.80455 3.48158 5.48614C2.81702 5.67161 2.21154 6.02538 1.72366 6.51325C1.23579 7.00113 0.882017 7.60661 0.696547 8.27117C0.223732 10.8248 -0.00936809 13.417 0.000288097 16.014C-0.00754264 18.6013 0.22555 21.1836 0.696547 23.7276C0.882017 24.3921 1.23579 24.9976 1.72366 25.4855C2.21154 25.9734 2.81702 26.3271 3.48158 26.5126C5.97317 27.1822 16.0142 27.1822 16.0142 27.1822C16.0142 27.1822 26.026 27.1822 28.5469 26.5126C29.2115 26.3271 29.8169 25.9734 30.3048 25.4855C30.7927 24.9976 31.1465 24.3921 31.3319 23.7276C31.7931 21.1827 32.0164 18.6003 31.9988 16.014C32.0182 13.418 31.7949 10.8257 31.3319 8.27117V8.27117ZM12.8104 20.8038V11.2096L21.1655 16.014L12.8104 20.8038Z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="css-0"
      >
        <img
          alt="NYPL Main Building Facade"
          className="css-0"
          src="https://cdn-d8.nypl.org/s3fs-public/2020-05/NYPL_MainFacadeRev2Cam2.png"
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <p>
        © The New York Public Library, 
        2025
      </p>
      <p>
        The New York Public Library is a 501(c)(3) | EIN 13-1887440
      </p>
    </div>
  </div>
  <div
    className="css-0"
  >
    <svg
      aria-hidden={false}
      aria-label="The New York Public Library"
      className="chakra-icon css-1vgifmd"
      fill="none"
      focusable={false}
      role="img"
      viewBox="0 0 201 165"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        The New York Public Library
      </title>
      <path
        d="M1.66113 3.03403H4.69144L26.7851 35.0843V3.03403H29.6501V40.1042H26.8402L4.52615 7.83329V40.1042H1.71623V3.03403H1.66113Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M46.8952 12.9635C39.8429 12.9635 35.4902 19.0867 35.4902 26.589C35.4902 34.2017 39.2368 40.711 47.3911 40.711C50.6115 40.7805 53.7723 39.8349 56.4269 38.008L55.2147 35.7463C52.9863 37.3598 50.3064 38.2284 47.5563 38.2286C41.1652 38.2286 38.6307 32.6571 38.5756 26.589V25.9822H56.8676C57.4186 18.4248 53.672 12.9635 46.8952 12.9635ZM46.785 15.5011C51.8539 15.5011 53.8924 19.5832 53.9475 23.6102H38.7409C39.5123 19.0867 42.1569 15.5011 46.785 15.5011Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M59.9531 13.5703H63.0936L69.5399 35.6911L76.482 13.5703H79.292L86.2341 35.7463L92.7355 13.5703H95.6556L87.6666 40.1594H85.022L77.8595 17.7628L70.6969 40.1594H68.0523L59.9531 13.5703Z"
        fill="white"
      />
      <path
        d="M121.22 26.1477L108.824 3.03403H112.295L122.818 23.0585H122.928L133.452 3.03403H136.592L124.251 26.1477V40.1042H121.22V26.1477Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M145.959 12.9635C137.639 12.9635 134.058 20.19 134.058 26.8649C134.058 33.5397 137.639 40.711 146.014 40.711C154.333 40.711 158.135 33.4294 158.135 26.7545C158.135 20.1348 154.278 12.9635 145.959 12.9635ZM145.793 15.4459C151.854 15.4459 155.05 20.9071 155.05 26.8649C155.05 32.8777 152.13 38.2286 146.014 38.2286C140.008 38.2286 137.033 32.7674 137.033 26.7545C137.033 20.7968 139.788 15.4459 145.793 15.4459Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M163.7 13.5703H166.62V18.2041C167.612 16.2734 170.036 12.9635 173.727 12.9635C174.592 12.957 175.45 13.1065 176.262 13.4048L175.545 16.1079C174.862 15.8362 174.132 15.7049 173.397 15.7217C169.705 15.7217 167.336 20.0797 166.62 21.6243V40.1042H163.7L163.7 13.5703Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M184.471 0H181.606V40.1042H184.471V0ZM196.372 13.5703L184.746 25.2099L196.978 40.1594H200.614L188.273 25.2099L200.008 13.5703H196.372Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M9.76031 58.86H1.71623H1.66113V95.9853H4.74653V80.2084H9.42973C17.8044 80.2084 22.4325 76.016 22.4325 69.3411C22.4325 62.7214 17.5289 58.86 9.76031 58.86ZM19.3471 69.4514C19.3471 74.3059 16.427 77.5605 9.37463 77.5605V77.6157H4.74653V61.5078H9.48483C16.5923 61.5078 19.3471 64.8177 19.3471 69.4514Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M27.1157 88.152V69.3963H30.0358V87.5452C30.0358 91.4618 31.4132 93.9442 35.4904 93.9442C39.4573 93.9442 42.708 90.1931 43.5896 88.6485V69.3963H46.4546V90.855C46.4253 92.5784 46.5917 94.2995 46.9504 95.9853H43.9752C43.7371 94.5445 43.6082 93.0876 43.5896 91.6273C42.3223 93.6132 39.292 96.5921 34.7741 96.5921C29.3196 96.5369 27.1157 92.8961 27.1157 88.152Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M53.7273 55.8259V94.551C55.5455 95.7646 58.5207 96.5369 61.6612 96.4818C69.7604 96.4818 74.7741 89.9724 74.7741 82.5253C74.7741 74.8023 71.0827 68.7343 63.5345 68.7343C60.8923 68.7539 58.3773 69.873 56.5923 71.8235V55.8259H53.7273ZM63.0386 71.3822C69.54 71.3822 71.7438 76.8986 71.7438 82.6908C71.7438 88.2072 68.4931 94.0545 61.7714 94.0545C59.989 94.0908 58.2207 93.7329 56.5923 93.0064V74.4162C58.2588 72.5953 60.5749 71.5052 63.0386 71.3822Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M80.7246 55.8259H83.6447V95.9301H80.7246V55.8259Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M92.6035 59.5972C91.9259 59.9927 91.5134 60.7226 91.5235 61.5078C91.4937 62.3058 91.9018 63.0562 92.5874 63.4641C93.2731 63.8719 94.1265 63.8719 94.8122 63.4641C95.4978 63.0562 95.906 62.3058 95.8761 61.5078C95.8863 60.7226 95.4737 59.9927 94.7962 59.5972C94.1186 59.2016 93.281 59.2016 92.6035 59.5972ZM95.215 69.3411H92.2398V95.9301H95.215V69.3411Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M100.78 82.6356C100.78 74.2507 105.628 68.7895 112.46 68.7895C116.427 68.7895 118.906 70.2789 120.339 71.5477L118.851 73.6439C117.07 72.0818 114.773 71.2366 112.405 71.2718C106.51 71.2718 103.865 76.4021 103.865 82.5253C103.865 88.4278 106.454 94.0545 112.35 94.0545C114.815 94.0736 117.21 93.2352 119.127 91.6825L120.394 93.8339C118.961 95.1027 116.151 96.5369 112.185 96.5369C104.526 96.5369 100.78 89.7518 100.78 82.6356Z"
        fill="white"
      />
      <path
        d="M0.889648 151.756V114.686H8.76844V145.412H22.5425V151.756H0.889648Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M28.936 119.798C29.7853 120.638 30.9356 121.101 32.1292 121.085C33.7427 121.144 35.2598 120.317 36.0842 118.927C36.9086 117.537 36.9086 115.807 36.0842 114.417C35.2598 113.027 33.7427 112.199 32.1292 112.259C30.9502 112.227 29.8077 112.671 28.9583 113.49C28.1089 114.309 27.6236 115.436 27.6113 116.617C27.6095 117.812 28.0867 118.958 28.936 119.798ZM28.4378 124.615V151.756H35.8758V124.615H28.4378Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M49.8153 126.767V111.652H42.4324V150.267C43.6996 150.873 47.4461 152.418 53.5619 152.418C61.7161 152.418 67.942 147.288 67.942 138.02C67.942 128.698 63.2037 124.009 56.4269 124.009C53.9527 124.052 51.5883 125.038 49.8153 126.767ZM53.5618 146.626C52.2865 146.653 51.0169 146.448 49.8153 146.019V131.897C51.0085 130.636 52.6538 129.901 54.3883 129.856C58.4103 129.856 60.1183 132.945 60.1183 138.02C60.1183 143.537 57.6941 146.626 53.5618 146.626Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M88.7133 131.125C87.8342 130.657 86.8413 130.447 85.8483 130.518C82.5425 130.518 80.6141 133.442 79.953 134.655V151.756H72.5701V124.615H79.953V128.311C81.11 126.325 83.369 123.953 86.7849 123.953C88.7684 123.953 89.8703 124.34 90.4764 124.671L88.7133 131.125Z"
        fill="white"
      />
      <path
        clipRule="evenodd"
        d="M107.887 148.832C107.936 149.819 108.084 150.799 108.328 151.756V151.701H115.6C115.049 149.071 114.79 146.389 114.829 143.702V133.938C114.829 126.712 111.303 123.898 103.92 123.898C98.1349 123.898 94.5536 126.381 92.9558 127.815L96.2065 132.338C98.1589 130.753 100.581 129.861 103.094 129.801C106.565 129.801 107.501 131.29 107.501 133.607V135.152H106.565C99.4021 135.152 92.1294 137.744 92.1294 144.805C92.1294 149.66 95.6556 152.418 100.284 152.418C103.22 152.392 105.997 151.082 107.887 148.832ZM106.51 139.73H107.446H107.501V144.64C106.249 146.117 104.422 146.981 102.488 147.012C100.78 147.012 99.1817 146.184 99.1817 144.199C99.1817 140.999 102.157 139.73 106.51 139.73Z"
        fill="white"
        fillRule="evenodd"
      />
      <path
        d="M136.868 131.125C136.006 130.663 135.033 130.452 134.058 130.518C130.752 130.518 128.824 133.442 128.162 134.655V151.756H120.78V124.615H128.162V128.311C129.319 126.325 131.578 123.953 134.994 123.953C136.978 123.953 138.08 124.34 138.686 124.671L136.868 131.125Z"
        fill="white"
      />
      <path
        d="M141.331 124.615H149.209L155.215 142.489H155.49L161 124.615H168.603L157.419 154.625C155.05 161.024 152.185 164.775 144.802 164.775C143.389 164.809 141.979 164.642 140.614 164.278L141.661 158.431C142.505 158.656 143.378 158.749 144.251 158.707C148.052 158.707 149.375 156.886 150.421 153.963L151.413 151.204L141.331 124.615Z"
        fill="white"
      />
    </svg>
  </div>
</footer>
`;
